proxy/cache_test.go

```diff
@@ -262,6 +262,74 @@ func TestCacheExpiration(t *testing.T) {
 	}
 }
 
+func TestCacheExpirationWithTTLOverride(t *testing.T) {
+	dnsProxy := createTestProxy(t, nil)
+	dnsProxy.CacheEnabled = true
+	dnsProxy.CacheMinTTL = 2
+	dnsProxy.CacheMaxTTL = 4
+	err := dnsProxy.Start()
+	if err != nil {
+		t.Fatalf("cannot start the DNS proxy: %s", err)
+	}
+
+	// Create dns messages with 1 second TTL
+	googleReply := dns.Msg{}
+	googleReply.SetQuestion("google.com.", dns.TypeA)
+	googleReply.Response = true
+	googleReply.Answer = []dns.RR{newRR("google.com. 1 IN A 8.8.8.8")}
+
+	youtubeReply := dns.Msg{}
+	youtubeReply.SetQuestion("youtube.com.", dns.TypeA)
+	youtubeReply.Response = true
+	youtubeReply.Answer = []dns.RR{newRR("youtube.com 60 IN A 173.194.221.198")}
+
+	dnsProxy.cache.Set(&youtubeReply)
+	dnsProxy.cache.Set(&googleReply)
+
+	// Wait a minimal amount of time, greater than the
+	// original minimum TTL entry.
+	time.Sleep(1100 * time.Millisecond)
+
+	//Update the expected youtube response TTL
+	//to be maxTTL (4) - time waited (1) = 3
+	youtubeReply.Answer[0].Header().Ttl = 3
+
+	r, ok := dnsProxy.cache.Get(&youtubeReply)
+	if !ok {
+		t.Fatalf("No cache found for %s", youtubeReply.Question[0].Name)
+	}
+	if diff := deepEqualMsg(r, &youtubeReply); diff != nil {
+		t.Error(diff)
+	}
+
+	r, ok = dnsProxy.cache.Get(&googleReply)
+	if !ok {
+		t.Fatalf("No cache found for %s", googleReply.Question[0].Name)
+	}
+	if diff := deepEqualMsg(r, &googleReply); diff != nil {
+		t.Error(diff)
+	}
+
+	// Wait an additional amount of time, greater than the max TTL,
+	// to guarrantee that all entries have been evicted.
+	time.Sleep(3100 * time.Millisecond)
```

Comment:
We should think about how to avoid sleeping for 3 seconds, otherwise we really slow down the tests.
Maybe we can just get entry from cache and check its TTL?

Reply:
That could be acceptable, if we are okay with not testing the cache's expiration functionality given the modified TTL's.

This is likely okay, considering that cache expiration is already tested in the existing `TestCacheExpiration` test.

That said, should I replace the additional wait, replacing it with just a check to ensure that the TTL returned matches our expectations?

Reply:
> should I replace the additional wait

I think so, yes.  TestCacheExpiration test should do its job - in this test we expect that auto-removing expired cache entries already works.

Reply:
> > should I replace the additional wait
> 
> I think so, yes. TestCacheExpiration test should do its job - in this test we expect that auto-removing expired cache entries already works.

Sounds good - I'll fix this in the next revision, along with the line length call out by the CI, as well as the `//TODO`